### PR TITLE
Clean code abort with invalid spin requested.

### PR DESCRIPTION
**Pull request by ye-luo**
a bit more than issue #168

### CODE REVIEW

**Comment by jtkrogel** [comment]
I would tell the user more about the problem in the error message, e.g. "Orbital H5 file contains no spin down data and is appropriate only for spin unpolarized calculations.  If this is your intent, please replace 'spindataset=1' with 'spindataset=0' in the input file."

Thanks for putting in the guard.

**Comment by ye-luo** [comment]
@jtkrogel agreed. The error message has been improved.

File: src/QMCWaveFunctions/EinsplineSetBuilderCommon.cpp [hunk 1]
```diff
@@ -769,8 +769,10 @@ EinsplineSetBuilder::OccupyBands(int spin, int sortBands, int numOrbs)
   if (myComm->rank() != 0) return;
   if(spin>=NumSpins)
   {
-    app_error() << "Invalid request for orbitals in spin group " << spin
-                << ". Current h5 file only contains spin groups " << "[0.." << NumSpins-1 << "]." << std::endl;
+    //app_error() << "Invalid request for orbitals in spin group " << spin
```
Comment (markdewing) [inline src/QMCWaveFunctions/EinsplineSetBuilderCommon.cpp]: Please remove commented-out code.

Comment (ye-luo) [inline src/QMCWaveFunctions/EinsplineSetBuilderCommon.cpp]: I prefer to keep it as a documentation line. The old error message is more for developer.

Comment (prckent) [inline src/QMCWaveFunctions/EinsplineSetBuilderCommon.cpp]: Please remove commented-out code. 

Put in a real comment or expand the error message to address user and developer.

Comment (ye-luo) [inline src/QMCWaveFunctions/EinsplineSetBuilderCommon.cpp]: Print both.

**Pull request merged**